app/parse.py

```diff
@@ -15,8 +28,154 @@ class Product:
+    def __init__(self) -> None:
+        chrome_options = webdriver.ChromeOptions()
+        chrome_options.add_argument("--headless")
+        self.driver = webdriver.Chrome(options=chrome_options)
+
+    def cookies_button(self) -> None:
+        while True:
+            try:
+                accept_cookies_button = (
+                    self.driver.find_element(By.CLASS_NAME, "acceptCookies")
+                )
+                accept_cookies_button.click()
+            except (
+                    NoSuchElementException,
+                    ElementClickInterceptedException,
+                    ElementNotInteractableException,
+                    TimeoutException
+            ):
+                break
+
+    def get_more_button(self) -> None:
+        while True:
+            try:
+                more_button = WebDriverWait(self.driver, 5).until(
+                    expected_conditions.element_to_be_clickable(
+                        (By.CLASS_NAME, "ecomerce-items-scroll-more")
+                    )
+                )
+                more_button.click()
+            except (
+                    ElementClickInterceptedException,
+                    ElementNotInteractableException,
+                    TimeoutException
+            ):
+                break
+
+    def get_page_of_products(self) -> list[Product]:
+        self.cookies_button()
+        self.get_more_button()
+
+        products = self.driver.find_elements(By.CLASS_NAME, "thumbnail")
+        products_list = []
+        for product in tqdm(products, desc="Processing"):
+            products_list.append(parse_single_product(product))
+        return products_list
+
+    def get_random_products_from_home(self) -> list[Product]:
+        self.driver.get(HOME_URL)
+        products_list = self.get_page_of_products()
+        return products_list
+
+    def get_random_computers(self) -> list[Product]:
+        url_computers = urljoin(HOME_URL, "computers/")
+        self.driver.get(url_computers)
+
+        computers_list = self.get_page_of_products()
+        return computers_list
+
+    def get_all_laptops(self) -> list[Product]:
+        url_laptops = urljoin(HOME_URL, "computers/laptops")
+        self.driver.get(url_laptops)
+
+        laptops_list = self.get_page_of_products()
+        return laptops_list
+
+    def get_all_tablets(self) -> list[Product]:
+        url_tablets = urljoin(HOME_URL, "computers/tablets")
+        self.driver.get(url_tablets)
+
+        tablets_list = self.get_page_of_products()
+        return tablets_list
+
+    def get_all_phones(self) -> list[Product]:
+        url_phones = urljoin(HOME_URL, "phones")
+        self.driver.get(url_phones)
+
+        phones_list = self.get_page_of_products()
+        return phones_list
+
+    def get_all_touch(self) -> list[Product]:
+        url_touch = urljoin(HOME_URL, "phones/touch")
+        self.driver.get(url_touch)
+
+        touch_list = self.get_page_of_products()
+        return touch_list
```

Comment:
Lots of duplicated functions, you can pass url in params and use just 1 function instead